vllm/distributed/kv_transfer/kv_connector/v1/shared_storage_connector.py

```diff
@@ -0,0 +1,362 @@
+        super().__init__(
+            rank=rank,
+            local_rank=local_rank,
+            config=config,
+            role=role,
+        )
+        self._block_size = config.cache_config.block_size
+        self._requests_need_load: list[str] = []
+        self._storage_path = config.kv_transfer_config.get_from_extra_config(
+            "shared_storage_path", "/tmp")
+        logger.info(config.kv_transfer_config)
+        logger.info("Shared storage path is %s", self._storage_path)
+
+    def start_load_kv(self, forward_context: "ForwardContext",
+                      **kwargs) -> None:
+        """Start loading the KV cache from the connector buffer to vLLM's 
+        paged KV buffer.
+
+        Args:
+            forward_context (ForwardContext): the forward context.
+            **kwargs: additional arguments for the load operation
+
+        Note:
+            The number of elements in kv_caches and layer_names should be 
+            the same.
+        """
+
+        def inject_kv_into_layer(
+            dst_kv_cache_layer: torch.Tensor,
+            src_kv_cache: torch.Tensor,
+            slot_mapping: torch.Tensor,
+        ) -> None:
+            """Inject the KV cache into the layer.
+
+            Args:
+                dst_kv_cache_layer (torch.Tensor): the destination KV cache 
+                    layer. In shape [2, num_pages, page_size, xxx].
+                src_kv_cache (torch.Tensor): the source KV cache. In shape
+                    [2, num_tokens, xxx].
+                slot_mapping (torch.Tensor): the slot mapping. In shape 
+                    [num_tokens].
+            """
+            dst_kv_cache_layer_shape = dst_kv_cache_layer.shape
+            num_pages = dst_kv_cache_layer_shape[1]
+            page_size = dst_kv_cache_layer_shape[2]
+            dst_kv_cache_layer = dst_kv_cache_layer.reshape(
+                2, num_pages * page_size, -1)
+            dst_kv_cache_layer[:, slot_mapping, ...] = src_kv_cache
+            dst_kv_cache_layer.reshape(dst_kv_cache_layer_shape)
+
+        # Get the metadata
+        metadata: KVConnectorMetadata = \
+            self._get_connector_metadata()
+        assert isinstance(metadata, SharedStorageConnectorMetadata)
+
+        if metadata is None:
+            logger.warning(
+                "In connector.start_load_kv, but the connector metadata is None"
+            )
+            return
+
+        attn_metadata = forward_context.attn_metadata
+        if attn_metadata is None:
+            logger.warning(
+                "In connector.start_load_kv, but the attn_metadata is None")
+            return
+
+        # Load the KV for each request each layer
+        for request in metadata.requests:
+            if request.is_store:
+                continue
+            logger.info("Inject KV cache of %d tokens to the paged memory",
+                        len(request.slot_mapping))
+            for layer_name in forward_context.no_compile_layers:
```

Comment:
does this mean we will have to disable `torch.compile()` with the actual production implementation?